src/server.ts

```diff
@@ -613,16 +749,33 @@ export class Server extends EventEmitter {
         const socket = this.connections.get(connectionId);
         if (!socket) return undefined;
 
+        // Socket contains application bytes only.
+        let srcTxBytes = socket.bytesWritten ?? 0;
+        let srcRxBytes = socket.bytesRead ?? 0;
+
+        if (this.serverType === 'https' && socket.tlsOverheadAvailable) {
+            /* eslint no-underscore-dangle: ["error", { "allow": ["_parent"] }] */
+            // Access underlying raw socket to get total bytes (app + TLS overhead).
+            const rawSocket = socket._parent;
+            if (rawSocket && typeof rawSocket.bytesWritten === 'number' && typeof rawSocket.bytesRead === 'number') {
+                if (rawSocket.bytesWritten >= socket.bytesWritten && rawSocket.bytesRead >= socket.bytesRead) {
+                    srcTxBytes = rawSocket.bytesWritten;
+                    srcRxBytes = rawSocket.bytesRead;
+                } else {
+                    // This should never happen, log for debugging.
+                    this.log(connectionId, `Warning: TLS overhead count error.`);
+                }
+            }
+        }
+
         const targetStats = getTargetStats(socket);
 
-        const result = {
-            srcTxBytes: socket.bytesWritten,
-            srcRxBytes: socket.bytesRead,
-            trgTxBytes: targetStats.bytesWritten,
-            trgRxBytes: targetStats.bytesRead,
+        return {
+            srcTxBytes, // HTTP: app only, HTTPS: total (app + TLS overhead)
+            srcRxBytes, // HTTP: app only, HTTPS: total (app + TLS overhead)
+            trgTxBytes: targetStats?.bytesWritten,
+            trgRxBytes: targetStats?.bytesRead,
         };
```

Comment:
Just an idea. Why don't we store the `_parent` socket in `this.connections`? Looking at the logic here in `getConnectionsStats`, you just fall back to the original socket for stats. That makes me question why would we ever want to use the TLS socket for connection tracking?
That gives me another idea, if you do `this.server.on('connection')` for HTTPS you should be able to reach the original Socket without using `_parent`, right?

Reply:
Ok, I agree